.github/workflows/release-starter.yml

```diff
@@ -0,0 +1,132 @@
+name: Release starter
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: The new version to release (`v2.16.0`)
+        required: true
+        type: string
+      pre-release:
+        description: Is this a pre-release ?
+        required: true
+        default: true
+        type: boolean
+
+# We set `concurrency` to prevent having this workflow being run multiple time for the same version
+concurrency:
+  group: release-starter-${{ inputs.version }}
+  cancel-in-progress: true
+
+jobs:
+  release-starter:
+    runs-on: ubuntu-22.04
+    name: Release starter 🚛
+    steps:
+      - name: Install python-semver-3.0.0
+        run: python3 -m pip install 'semver~=3.0.0'
+        timeout-minutes: 5
+
+      - name: Parse version `${{ inputs.version }}`
+        id: version
+        shell: python
+        run: |
+          import os
+
+          import semver
+
+          raw_version = os.environ["VERSION"]
+
+          if raw_version.startswith("v"):
+              raw_version = raw_version[1:]
+
+          version = semver.Version.parse(raw_version, optional_minor_and_patch=False)
+
+          with open(os.environ["GITHUB_OUTPUT"], "a") as f:
+              f.write(
+                  "\n".join(
+                      [
+                          f"full={version}",
+                          f"major={version.major}",
+                          f"minor={version.minor}",
+                          f"patch={version.patch}",
+                          f"prerelease={version.prerelease or ''}",
+                          f"build={version.build}",
+                      ]
+                  )
+              )
+        env:
+          VERSION: ${{ inputs.version }}
+
+      - name: Checkout the repo at base `${{ github.ref }}`
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # pin v3.5.2
+        timeout-minutes: 5
+
+      - name: Release branch & tag name
+        id: names
+        run: |
+          echo "branch=release/${{ steps.version.outputs.major }}.${{ steps.version.outputs.minor }}" >> $GITHUB_OUTPUT
+          echo "tag=v${{ steps.version.outputs.full }}" >> $GITHUB_OUTPUT
+
+      - name: Start summary
+        run: |
+          (
+            echo '# Parsec ${{ steps.names.outputs.tag }}'
+            echo
+            echo 'Initiated by: @${{ github.triggering_actor }}'
+            echo 'Base branch: `${{ github.ref }}`'
+            echo 'Target branch: `${{ steps.names.outputs.branch }}`'
+            echo
+          ) >> "$GITHUB_STEP_SUMMARY"
+
+      - name: Switch or create git branch
+        run: >-
+          git switch ${{ steps.names.outputs.branch }}
+          || git switch --create ${{ steps.names.outputs.branch }}
+          git push --set-upstream origin origin ${{ steps.names.outputs.branch }}
```

Comment:
this is a noop if the branch already exists (hence no need to do a if) ?

if so, a comment would be welcome